crates/biome_js_analyze/src/analyzers/nursery/no_useless_lone_block_statements.rs

```diff
@@ -150,7 +150,16 @@ fn in_control_structure(block: &JsBlockStatement) -> bool {
         if JsElseClause::can_cast(syntax_kind) {
             return true;
         }
+
+        if JsCatchClause::can_cast(syntax_kind) {
+            return true;
+        }
+
+        if JsFinallyClause::can_cast(syntax_kind) {
+            return true;
+        }
```

Comment:
I think we could simplify to:

```suggestion
        matches!(syntax_kind, JsSYntaxKind::JS_ELSE_CLAUSE | JsSYntaxKind::JS_CATCH_CLAUSE | JsSYntaxKind::JS_FINALLY_CLAUSE)
```

You could even use:

```rs
    matches!(block.syntax().parent().kind(), Some(JsSYntaxKind::JS_ELSE_CLAUSE | JsSYntaxKind::JS_CATCH_CLAUSE | JsSYntaxKind::JS_FINALLY_CLAUSE))
```

Reply:
Since macros are no longer needed, I've made the following modifications.

```rust
if let JsSyntaxKind::JS_ELSE_CLAUSE
| JsSyntaxKind::JS_CATCH_CLAUSE
| JsSyntaxKind::JS_FINALLY_CLAUSE = syntax_kind
{
    return true;
}
```

Reply:
> Since macros are no longer needed

Oh interesting. Have you a reference for context about that?

Reply:
[This](https://doc.rust-lang.org/rust-by-example/flow_control/if_let.html) is the documentation for `if-let`, but it seems that the example you mentioned is not referred to. The relevant implementation of the parser can be found here, and as written in the comments, it's possible to write a pattern after `let` just like in a `match` statement.

https://github.com/rust-lang/rust/blob/a04d56b36d8f634abd7bdd64dd859a30655f1818/compiler/rustc_parse/src/parser/expr.rs#L2491

Reply:
Ok I see. I am not sure it is more idiomatic to use `if let` than `matches!()` in this case.

Thanks for your contribution :)

Reply:
In my opinion, the `matches!` macro is useful for simplifying expressions like `matches!(foo, Some(bar) if bar == 2)` that require an `if` statement. Other than that, `if let` seems to be used more often. This is also the case in the Rust compiler's source code, where `if let` is often used for simple pattern matching.